### ISSUE TITLE
Private Repo Access

### ISSUE THREAD

**Issue by jnylen** [pmac/dokku-webhook-deploy#4]
Hi @pmac,

I'm currently setting up Dokku with this plugin on some of my projects and it's working lovely so far, thanks for the amazing work!

One issue i'm encountering currently is that if the repo is a private repo then it's unable to access the repository through the http link that it receives from the body at [this line](https://github.com/pmac/dokku-webhook-deploy/blob/master/webhook_deploy/dokku.py#L89). Is this something you know about?

What I would assume would be nice to have is a config environment variable that has a TRUE/FALSE variable that then does the pull via ssh instead.
Basically just grab everything after `github.com/` and add it to `git@github.com:`.

**Comment by pmac** [comment]
Hi! I'm glad it's working for you. That is a use case I didn't have yet so I haven't ensured it works. So you're saying that you''d get your SSH key into the docker image somehow and then the SSH Github URL would work? That is one way to do it. Another would be to generate a personal access token and then we could add it as a dokku environment config option that we could then insert into the Github clone URL. But I guess the SSH dir from the server is mounted right? So maybe the SSH option is easiest. What do you think?

**Comment by jnylen** [comment]
Hi @pmac,

I think Github Access Token is probably the easiest. That way you can just do this:
`git clone https://<token>:x-oauth-basic@github.com/owner/repo.git`

Also thanks for answering, I would've done this myself but sadly I don't know anything about python.

**Comment by pmac** [comment]
No worries. I'll see if I can find some time soon to make that happen. It should be a pretty easy change.

**Comment by ItzNeviKat** [comment]
Any updates?
